frontend/src/lib/components/PropertyFilters/PropertyValue.js

```diff
@@ -32,15 +32,15 @@ export function PropertyValue({
             })
             setOptionsCache({ ...optionsCache, [value]: true })
         } else {
-            api.get(endpoint || 'api/' + type + '/values/?key=' + key + (value ? '&value=' + value : '')).then(
-                (propValues) => {
-                    setOptions({
-                        [propertyKey]: { values: [...new Set([...propValues.map((option) => option)])], status: true },
-                        ...options,
-                    })
-                    setOptionsCache({ ...optionsCache, [value]: true })
-                }
-            )
+            api.get(
+                endpoint || 'api/projects/@current/' + type + 's/values/?key=' + key + (value ? '&value=' + value : '')
```

Comment:
Sneaking the `s` on here is both tricky and seems like not really a good idea - seems like it would lead to unexpected brokedness.

One thing that might help here is converting the file to typescript and setting `type` argument to a strict union, e.g. `type: 'action' | 'cohort' | ...` and making sure the type is accurate to life?

Alternatively, updating all the callpoints - which is less effort?

Reply:
I refactored the file to TypeScript… but that actually didn't help much, since this logic is quite convoluted and spread over at several layers (PropertyFilters.js > PropertyFilter.js > PropertyValue.js). I just added a comment to explain the sneaky `s` here and I honestly think that's the safest solution for now, leaving all the `'cohort'` etc. values the same.